search-commons/src/main/java/no/unit/nva/search2/model/ParameterKey.java

```diff
@@ -26,7 +28,7 @@ static Predicate<ParameterKey> equalTo(String name) {
     }
 
     static String escapeSearchString(String value) {
-        return value.replaceAll("([-+&|!(){}\\[\\]^\"~*?:\\\\/])", "\\\\$1");
+        return value.replaceAll(PATTERN_IS_SPECIAL_CHARACTERS, PATTERN_IS_ADD_SLASH);
```

Comment:
Is this the magic strings solving the task?

Reply:
This is i list of charcters that requiers an added escape '\'  inorder to work in opensearch.

( [ &! ] ) -> search for '&' or '!', when encountered add to (a) find group, and replace with '\+(fond group)'
anyone of these characters should be replaced by  '/' + (character)

If you have better names or a solutions that doesn't requires 50 lines of code, and doesn't have to iterate text for each char,   I am listening